server/impl/src/main/java/com/walmartlabs/concord/server/security/ldap/LdapUserInfoProvider.java

```diff
@@ -51,6 +57,12 @@ public UserType getUserType() {
 
     @Override
     public UserInfo getInfo(UUID id, String username, String userDomain) {
+        /* Get User Data from database when user logged in via SSO */
+        UserPrincipal u = UserPrincipal.getCurrent();
+        if (u != null && u.getRealm().equals(SSO_REALM_NAME)){
```

Comment:
This is an abstraction leak. LDAP auth provider should not know anything about other authentication providers.
I don't have a better idea right now, but this is definitely not how it should be.

Same goes for [this](https://github.com/walmartlabs/concord/blob/2de2887476d8973d44dc0757f579b5382e14d5b3/server/impl/src/main/java/com/walmartlabs/concord/server/user/UserManager.java#L104) and [this](https://github.com/walmartlabs/concord/blob/master/server/impl/src/main/java/com/walmartlabs/concord/server/user/UserManager.java#L150) - I've already merged it, but it's more or less the same amount of "leakage" :-)

I'll ponder on a better way.

Reply:
Ok.. how about having another class SsoUserInfoProvider implementation of UserInfoProvider to keep abstraction?

Reply:
Yes, sounds like a good idea.

Reply:
refactored